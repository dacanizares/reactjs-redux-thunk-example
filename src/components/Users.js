import React from 'react'
import User from './User'
import { Row, Col, CardDeck } from 'reactstrap'

const Users = ({usersList, fetchUsers}) =>  {   
  return (
    <>
      <Row>
        <CardDeck>
          {usersList.map((user) => 
            <User 
              email={user.email} 
              name={`${user.name.title} ${user.name.last}`} 
              avatarUrl={user.picture.large} 
              key={user.id.value} />    
          )}
        </CardDeck>
      </Row>
      
      <Row className='mt-3'>
        <Col xs='12'>
          <button 
            className='btn btn-primary btn-lg' 
            onClick={() => fetchUsers()} >
            Update
          </button>
        </Col>
      </Row>
    </>
  );
}

export default Users;