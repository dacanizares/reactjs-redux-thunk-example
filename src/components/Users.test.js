import React from 'react'
import Users from './Users'
import renderer from 'react-test-renderer'

jest.mock('./User', () => 'User');

it('renders correctly', () => {
  const fecth = () => { }

  const userList = [{
    email: 'email@test.com', 
    name: {
      title: 'mr',
      first: 'name'
    },
    picture: {
      large: 'image here'
    },
    id: {
      value: 'uniquevalue'
    }
  }];

  const tree = renderer
    .create(<Users usersData={userList} fetchData={fecth} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('fails when no wrong type of data was given', () => {
  expect(() => renderer
  .create(<Users usersData={{}} />)).toThrowError();
});