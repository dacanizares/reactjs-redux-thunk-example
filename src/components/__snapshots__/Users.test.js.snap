// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="card-deck"
    >
      <User
        avatarUrl="image here"
        email="email@test.com"
        name="mr undefined"
      />
    </div>
  </div>,
  <div
    className="mt-3 row"
  >
    <div
      className="col-12"
    >
      <button
        className="btn btn-primary btn-lg"
        onClick={[Function]}
      >
        Update
      </button>
    </div>
  </div>,
]
`;
